### PR TITLE
Assessment grading view

### PR DESCRIPTION
**Pull request by pran-sada**
### Description

- Submitted assessment details view with rubrics and multiple attempts

#### Video

https://user-images.githubusercontent.com/116863369/217245933-b4b0b419-fd97-4101-bbc7-77ef9cf9eb05.mp4

### Other Details

- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-2184)
- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-2185)
- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-2186)

#### Checklist

- [ ] **Assumptions of User Story met (including acceptance criteria)**
- [ ] **All new necessary Unit tests were CREATED** to promote code coverage
- [ ] **All required unit tests are PASSING**
- [ ] The proposed functionality was tested manually locally, and it's working fine.
- [ ] PR was properly reviewed by a team member (at least 1 approval required to merge!)
- [ ] Unnecessary console.logs AND/OR comments were removed
- [ ] Feature ok-ed by UX designer (If needed - not for back end components) and Product Owner
- [ ] Ensure configuration and/or build changes documented

### CODE REVIEW

File: apps/toboggan-app/src/app/assessment/interface/assessments.type.ts [hunk 1]
```diff
@@ -89,3 +89,44 @@ export interface IAssessmentFlag {
...
+export enum RuleEngineNodeType {
+  LearningExperience = 'learning_experiences',
+  CurriculamPathways = 'curriculum_pathways',
+  LearningObjects = 'learning_objects',
+  LearningResources = 'learning_resources',
+  Assessments = 'assessments',
+  SubmittedAssessments = 'submitted_assessments',
+}
+
+export enum RuleEngineVerb {
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/assessment/interface/assessments.type.ts:110]: keep items commonly used for rule engine for api and UI in toboggan-common

**Review by sada-ajith — CHANGES_REQUESTED** [review]

File: apps/toboggan-api/src/app/modules/assessments/assessments.service.ts [hunk 2]
```diff
@@ -32,12 +25,25 @@ export class AssessmentsService {
...
   }
 
-  getAssessmentsById(assessor_id:string) {
-    return this.httpService.get(`/submitted-assessments?assessor_id=${assessor_id}&status=non_evaluated`) 
+  getAssessmentsById(
+    assessor_id: string,
+    status: SubmittedAssessmentFilterStatus
+  ) {
+    return this.httpService.get(
+      `/submitted-assessments?assessor_id=${assessor_id}&status=${status}`
```
Comment (shafi-sadaindia) [inline apps/toboggan-api/src/app/modules/assessments/assessments.service.ts:33]: Just a suggestion: shall we generate url params from object?.
ex : getAssessmentsById({assessor_id: id, status:'status'});

File: apps/toboggan-api/src/app/modules/assessments/assessments.controller.ts [hunk 3]
```diff
@@ -3,46 +3,107 @@ import {
...
+  @Put('/submittedAssessment/:uuid')
   updateSubmittedAssessment(
-    @Param('uuid') uuid,
+    @Param('uuid') uuid: string,
     @Body() body: Partial<ISubmittedAssessment>
   ) {
     return this.assessmentsService.updateSubmittedAssessment(uuid, body);
   }
+
+  @Post('/learnosity/init/assessment') //! Temporary endpoint
```
[the remark this answers is not in the document]
Comment (pran-sada) [reply apps/toboggan-api/src/app/modules/assessments/assessments.controller.ts]: Removed

File: apps/toboggan-api/src/app/app.module.ts [hunk 4]
```diff
@@ -9,17 +11,37 @@ import { AuthenticationModule } from './modules/authentication/authentication.mo
...
+    ConfigModule.forRoot(),
     AuthModule,
     UsersModule,
     GroupsModule,
     PermissionsModule,
     AssessmentsModule,
     AuthenticationModule,
+    LearnosityModule,
+    LRSModule,
+    TypeOrmModule.forRoot({
```
[the remark this answers is not in the document]
Comment (pran-sada) [reply apps/toboggan-api/src/app/app.module.ts]: Removed the unwanted modules

File: apps/toboggan-api/src/app/app.module.ts [hunk 5]
```diff
@@ -9,17 +11,37 @@ import { AuthenticationModule } from './modules/authentication/authentication.mo
...
 @Module({
   imports: [
+    ConfigModule.forRoot(),
     AuthModule,
     UsersModule,
     GroupsModule,
     PermissionsModule,
     AssessmentsModule,
     AuthenticationModule,
+    LearnosityModule,
```
[the remark this answers is not in the document]
Comment (pran-sada) [reply apps/toboggan-api/src/app/app.module.ts]: Removed unwanted modules

File: apps/toboggan-api/src/app/modules/rule-engine/rule-engine.dto.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { IsEnum, IsNotEmpty, IsString } from 'class-validator';
+import { RuleEngineNodeType, RuleEngineVerb } from './rule-engine.types';
```
[the remark this answers is not in the document]
Comment (pran-sada) [reply apps/toboggan-api/src/app/modules/rule-engine/rule-engine.dto.ts]: Renaming done

File: apps/toboggan-app/src/app/assessment/components/assessment-list/assessment-list.component.ts [hunk 7]
```diff
@@ -172,6 +195,16 @@ export class AssessmentListComponent implements OnInit, OnDestroy {
             1: cellData.maxAttempts,
             cellClass: attemptBorderCellClass,
           },
+          // instructor: cellData.instructor,
```
[the remark this answers is not in the document]
Comment (pran-sada) [reply apps/toboggan-app/src/app/assessment/components/assessment-list/assessment-list.component.ts]: Removed the commented code

File: apps/toboggan-app/src/app/assessment/components/assessment-list/assessment-table.type.ts [hunk 8]
```diff
@@ -61,12 +61,5 @@ export const assessmentTableHeader: TableColumnDisplayMetadatum[] = [
     searchableField: true,
     alignment: TableColumnAlignmentEnum.Left,
     dataType: TableColumnDataTypeEnum.TextNowrap,
-  }
+  },
 ];
-
-export enum RowActions {
```
[the remark this answers is not in the document]
Comment (pran-sada) [reply apps/toboggan-app/src/app/assessment/components/assessment-list/assessment-table.type.ts:67]: This was moved to another file where we stored the enum values

File: apps/toboggan-app/src/app/assessment/components/assessment-submission-details/assessment-submission-details.component.ts [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+    this.injectServiceSubscription =
+      this.learnosityItemsInjectService.renderItemsComplete$.subscribe(
+        ({ itemsApp }) => {
+          this.itemsAppObj = itemsApp;
+        }
+      );
+  }
+
+  previousAssessment(attemptNo: number) {
+    if (attemptNo <= 1) return;
```
[the remark this answers is not in the document]
Comment (pran-sada) [reply apps/toboggan-app/src/app/assessment/components/assessment-submission-details/assessment-submission-details.component.ts]: Made it to a common function

File: apps/toboggan-app/src/app/assessment/interface/assessments.type.ts (hunk 1 above)
Comment (pran-sada) [reply apps/toboggan-app/src/app/assessment/interface/assessments.type.ts:110]: Nest use snake case and Angular uses camel case, hence the key values will be different

File: apps/toboggan-api/src/app/modules/assessments/assessments.service.ts (hunk 2 above)
Comment (pran-sada) [reply apps/toboggan-api/src/app/modules/assessments/assessments.service.ts:33]: These are query params not passed with body

Comment (shafi-sadaindia) [reply apps/toboggan-api/src/app/modules/assessments/assessments.service.ts:33]: Nop. we can add more params by converting an object to query params.

**Review by sada-ajith — APPROVED** [review]

**Pull request merged**